lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/AllocatedEvaluator.java

```diff
@@ -96,4 +96,16 @@ void submitContextAndTask(final Configuration contextConfiguration,
   void submitContextAndServiceAndTask(final Configuration contextConfiguration,
                                       final Configuration serviceConfiguration,
                                       final Configuration taskConfiguration);
+
+  /**
+   * Check whether this evaluator is in closing state.
+   * @return whether this evaluator is in closing state.
+   */
+  boolean isClosing();
+
+  /**
+   * Check whether this evaluator is in closed state.
+   * @return whether this evaluator is in closed state.
+   */
+  boolean isClosed();
```

Comment:
Why do we expose `isClosed` and `isClosing` in the `AllocatedEvaluator` ?  Who calls this methods ?